### PR TITLE
fix(md-tooltip): removed watcher for performance

### PR DESCRIPTION
**Pull request by clshortfuse**
md-tooltip uses a watcher per tooltip. The watcher is just to watch if scope.visible changes. This variable only changes internally so there's no reason to need a watcher when the changes could be tracked manually.

fixes https://github.com/angular/material/issues/4033

### CODE REVIEW

**Comment by clshortfuse** [comment]
Okay. I see the issue. md-visible can change scope.visible from the outside. I was wrong to assume that .visible as just an internal value. That's really the only issue left.

File: src/components/tooltip/tooltip.js [hunk 1]
```diff
@@ -81,14 +82,20 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdThe
 
     function configureWatchers () {
       scope.$on('$destroy', function() {
-        scope.visible = false;
+        if (scope.visible !== false) {
+          scope.visible = false;
+          $mdUtil.nextTick(checkVisibility);
```
Comment (ThomasBurleson) [inline src/components/tooltip/tooltip.js]: Why are you waiting a frame and then calling hideTooltip when the scope has been destroyed?

**Comment by ThomasBurleson** [comment]
@clshortfuse - please provide 2-n `it(...)` tests that validate these changes.

**Comment by clshortfuse** [comment]
That was me trying to exactly replicate the $watch behavior trying to figure out why the tests failed. Then I realized it was because md-visible.

**Comment by ThomasBurleson** [comment]
@clshortfuse - should I close this PR without trying merge? If no, then please squash and rebase.

**Comment by clshortfuse** [comment]
It's not right and buggy. I'll have to revisit it another day. For now, I'll close it.

**Pull request closed without merging**